src/django_pg_migration_tools/management/commands/migrate_with_timeouts.py

```diff
@@ -28,32 +54,304 @@ def add_arguments(self, parser: base.CommandParser) -> None:
+            dest="statement_timeout_retry_max_wait_in_ms",
+            type=int,
+            required=False,
+            default=10_000,
+            help=(
+                "Sets a limit to the length of time between subsequent "
+                "exponential backoff propagation retries. Defaults to 10s."
+            ),
+        )
+        parser.add_argument(
+            "--statement-timeout-retry-min-wait-in-ms",
+            dest="statement_timeout_retry_min_wait_in_ms",
+            type=int,
+            required=False,
+            default=3_000,
+            help=(
+                "Sets a limit to the minimum length of time between subsequent "
+                "exponential backoff propagation retries. Defaults to 3s."
+            ),
+        )
         super().add_arguments(parser)
 
     @base.no_translations
     def handle(self, *args: Any, **options: Any) -> None:
-        statement_timeout_in_ms: int | None = options["statement_timeout_in_ms"]
-        lock_timeout_in_ms: int | None = options["lock_timeout_in_ms"]
+        self._initialise(**options)
+        self._validate()
+        self._migrate(*args, **options)
 
-        if statement_timeout_in_ms is None and lock_timeout_in_ms is None:
+    def _initialise(self, **options: Any) -> None:
+        self.statement_timeout = _parse_non_required_ms_option_to_timedelta(
+            options, key="statement_timeout_in_ms"
+        )
+        self.statement_timeouts = 0
+        self.statement_timeout_max_retries = options.pop(
+            "statement_timeout_max_retries"
+        )
+        self.statement_timeout_retry_exp = options.pop("statement_timeout_retry_exp")
+        self.statement_timeout_retry_max_wait = _parse_int_in_ms_to_timedelta(
+            options.pop("statement_timeout_retry_max_wait_in_ms")
+        )
+        self.statement_timeout_retry_min_wait = _parse_int_in_ms_to_timedelta(
+            options.pop("statement_timeout_retry_min_wait_in_ms")
+        )
+        self.lock_timeout = _parse_non_required_ms_option_to_timedelta(
+            options, key="lock_timeout_in_ms"
+        )
+        self.lock_timeouts = 0
+        self.lock_timeout_max_retries = options.pop("lock_timeout_max_retries")
+        self.lock_timeout_retry_exp = options.pop("lock_timeout_retry_exp")
+        self.lock_timeout_retry_max_wait = _parse_int_in_ms_to_timedelta(
+            options.pop("lock_timeout_retry_max_wait_in_ms")
+        )
+        self.lock_timeout_retry_min_wait = _parse_int_in_ms_to_timedelta(
+            options.pop("lock_timeout_retry_min_wait_in_ms")
+        )
+        self.retry_callback = None
+        if callback_path := options.pop("retry_callback_path", None):
+            # This raises ModuleNotFoundError, which gives a good explanation
+            # of the error already (see tests). We don't have to wrap this into
+            # our own exception.
+            *module, callback = callback_path.split(".")
```

Comment:
in my suggestion I do a `module, attr_name = callaback_path.rsplit(".")` here to avoid needing the ".".join(module). I also make sure the path has at least one dot in it and I `assert callable(callback)`